vllm/model_executor/models/pixtral.py

```diff
@@ -788,20 +791,25 @@ def __init__(
         super().__init__()
 
         assert config.intermediate_size is not None
-        # TODO: Use quant_config and prefix after optimizing this
-        self.gate_proj = nn.Linear(config.hidden_size,
-                                   config.intermediate_size,
-                                   bias=False)
-        self.up_proj = nn.Linear(config.hidden_size,
-                                 config.intermediate_size,
-                                 bias=False)
-        self.down_proj = nn.Linear(config.intermediate_size,
-                                   config.hidden_size,
-                                   bias=False)
+        self.gate_up_proj = MergedColumnParallelLinear(
+            input_size=config.hidden_size,
+            output_sizes=[config.intermediate_size] * 2,
+            bias=False,
+            quant_config=quant_config,
+            prefix=f"{prefix}.gate_up_proj")
+        self.down_proj = RowParallelLinear(input_size=config.intermediate_size,
+                                           output_size=config.hidden_size,
+                                           bias=False,
+                                           quant_config=quant_config,
+                                           prefix=f"{prefix}.down_proj")
         self.act = get_act_fn(config.hidden_act)
 
     def forward(self, x: torch.Tensor) -> torch.Tensor:
-        return self.down_proj(self.act(self.gate_proj(x)) * self.up_proj(x))
+        gate_up, _ = self.gate_up_proj(x)
+        d = gate_up.shape[-1] // 2
+        x = self.act(gate_up[..., :d]) * gate_up[..., d:]
+        x, _ = self.down_proj(x)
+        return x
```

Comment:
Can we use the `act_and_mul` kernel?

Reply:
There are some pixtral checkpoints out there that use gelu, so I didn't want to make those stop working by only supporting SiluAndMul

See some models here:
* https://huggingface.co/SeanScripts/pixtral-12b-nf4/blob/ce4c79e0a2d82b2f8f4ff66a65c95cd4564ef492/config.json#L128
* https://huggingface.co/Ertugrul/Pixtral-12B-Captioner-Relaxed/blob/768ab0dc91f78088f523167f02158877a5f0b240/config.json#L113

Reply:
Maybe not the PR but there is a `GeluAndMul` module,

why not make a `get_act_and_mul_fn` to replace `get_act_fn` in the init?

Reply:
This is a fair point, I can make this